### PR TITLE
Fix PipelineTests skip conditions

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

While I am starting to try to fix some pipeline tests that are skipped previously, I realized PR #21516 (removing the usage of meta class in pipeline testing) **accidentally skip more tests than it should**.

- previously, each combination of model/tokenizer/processor class has its own test case (being generated on the fly), so we can use `self.skipTest` to skip some cases.
- After #21516，each model + task has its test cases, but inside that test, it runs against all tokenizers/processors that are available. In particular, slow / fast tokenizer. 
  - As mentioned before, we have some slow tokenizer issues in pipeline testing (never tested before #20426), and we skip some failing cases for now.
  - but after #21516, when we use `self.skipTest`, we **might** **skip the next combination(s)**. This is **NOT good/expected**.

This PR instead uses `logger.warning` to log the skipped cases and continue to the next combination in a test. This is not very pleasant, but as we don't want to use meta class, this is the only way I can think of for now.

It's better to prepare a report file that clearly indicate which test cases and/or combination being [skipped].

### CODE REVIEW

File: tests/models/luke/test_modeling_luke.py [hunk 1]
```diff
@@ -619,6 +619,15 @@ class LukeModelTest(ModelTesterMixin, PipelineTesterMixin, unittest.TestCase):
...
     test_head_masking = True
 
+    # TODO: Fix the failed tests
+    def is_pipeline_test_to_skip(
+        self, pipeline_test_casse_name, config_class, model_architecture, tokenizer_name, processor_name
+    ):
+        if pipeline_test_casse_name in ["QAPipelineTests", "ZeroShotClassificationPipelineTests"]:
+            return True
+
+        return False
```
Comment (ydshieh) [inline tests/models/luke/test_modeling_luke.py:629]: We need this due to the change in `tests/test_pipeline_mixin.py` (the change at the end of that file).
Same for the changes in the following model test files.

File: tests/test_pipeline_mixin.py [hunk 2]
```diff
@@ -179,11 +181,12 @@ def run_model_pipeline_tests(self, task, repo_name, model_architecture, tokenize
                     tokenizer_name,
                     processor_name,
                 ):
-                    self.skipTest(
+                    logger.warning(
```
Comment (ydshieh) [inline tests/test_pipeline_mixin.py:184]: We should NOT skip at this moment - as we might skip the next combination of tokenizer/processor.

File: tests/test_pipeline_mixin.py [hunk 3]
```diff
@@ -429,10 +436,3 @@ def validate_test_components(test_case, task, model, tokenizer, processor):
             raise ValueError(
                 "Could not determine `vocab_size` from model configuration while `tokenizer` is not `None`."
             )
-        # TODO: Remove tiny models from the Hub which have problematic tokenizers (but still keep this block)
-        if config_vocab_size is not None and len(tokenizer) > config_vocab_size:
-            test_case.skipTest(
-                f"{test_case.__class__.__name__}::test_pipeline_{task.replace('-', '_')} is skipped: tokenizer "
-                f"(`{tokenizer.__class__.__name__}`) has {len(tokenizer)} tokens which is greater than "
-                f"`config_vocab_size` ({config_vocab_size}). Something is wrong."
-            )
```
Comment (ydshieh) [inline tests/test_pipeline_mixin.py:438]: Shout not use `skipTest`.

**Review by sgugger — APPROVED** [review]
LGTM, thanks!

**Pull request merged**